### PR TITLE
Import smart survey wdyh data

### PR DESCRIPTION
**Pull request by dwhenry**

### CODE REVIEW

File: app/models/where_did_you_hear.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@ class WhereDidYouHear < ActiveRecord::Base
   before_validation :lookup_mappings
 
   def lookup_mappings
-    self.heard_from       = CodeLookup.for(value: heard_from_code)
-    self.pension_provider = CodeLookup.for(value: pension_provider_code)
+    %w(heard_from pension_provider).each do |field|
```
Comment (benlovell) [inline app/models/where_did_you_hear.rb]: I don't think this is any clearer / more concise than the code it replaced, to be fair.

File: lib/importers/smart_survey/call_record.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      def pension_provider
+        @row[10].to_s == '-' ? '' : @row[10].to_s
+      end
+
+      def valid?
+        user_id.to_i.to_s == user_id
+      end
+
+      def self.build(io, delivery_partner)
+        rows = CSV.parse(remove_bom(io.read))
```
Comment (benlovell) [inline lib/importers/smart_survey/call_record.rb]: I'm not 100% on this, but maybe you could specify the `encoding: 'bom|utf-8'` option on `.parse` here to remove the need for stripping the byte-order mark?

**Comment by benlovell** [comment]
Needs a rebase.

File: lib/importers/smart_survey/call_record.rb (hunk 2 above)
Comment (dwhenry) [inline lib/importers/smart_survey/call_record.rb]: The works by setting the encode when the file is opened. As this is a StringIO object would need to write to a temporary file and then read it in again for that to work.

File: app/models/where_did_you_hear.rb (hunk 1 above)
Comment (dwhenry) [inline app/models/where_did_you_hear.rb]: it adds `unless self[field].present?` at the end

**Comment by dwhenry** [comment]
rebase done. 🍎

File: app/models/where_did_you_hear.rb [hunk 3]
```diff
@@ -5,4 +5,12 @@ def lookup_mappings
     self.heard_from       = CodeLookup.for(value: heard_from_code)
     self.pension_provider = CodeLookup.for(value: pension_provider_code)
   end
+
+  def heard_from=(*)
+    heard_from.presence || super
```
Comment (benlovell) [inline app/models/where_did_you_hear.rb]: I don't use `presence` enough. Nice. 👍

**Comment by benlovell** [comment]
:shipit:

**Pull request merged**